src/test/java/uk/gov/pay/products/persistence/dao/PaymentDaoIT.java

```diff
@@ -36,7 +36,34 @@ public void before(){
     }
 
     @Test
-    public void shouldSuccess_whenFindingAValidPayment() {
+    public void shouldFindAPaymentByGovukPaymentId() {
+        PaymentEntity payment = PaymentEntityFixture.aPaymentEntity()
+                .withGovukPaymentId(randomUuid())
+                .withStatus(PaymentStatus.CREATED)
+                .withProduct(productEntity)
+                .withReferenceNumber("MH2KJY5KPW")
+                .build();
+        databaseHelper.addPayment(payment.toPayment(), 1);
+
+        Optional<PaymentEntity> paymentEntity = paymentDao.findByGovukPaymentId(payment.getGovukPaymentId());
+        assertTrue(paymentEntity.isPresent());
+
+        assertThat(paymentEntity.get().getExternalId(), is(payment.getExternalId()));
+        assertThat(paymentEntity.get().getNextUrl(), is(payment.getNextUrl()));
+        assertThat(paymentEntity.get().getAmount(), is(payment.getAmount()));
+        assertThat(paymentEntity.get().getStatus(), is(payment.getStatus()));
+        assertThat(paymentEntity.get().getGovukPaymentId(), is(payment.getGovukPaymentId()));
+        assertNotNull(paymentEntity.get().getDateCreated());
+    }
+    
+    @Test
+    public void shouldNotFindAPaymentByGovukPaymentId() {
+        Optional<PaymentEntity> paymentEntity = paymentDao.findByGovukPaymentId(randomUuid());
+        assertTrue(paymentEntity.isEmpty());
+    }
```

Comment:
I meant have a test where there’s an entity with a non-matching ID in the database rather than just searching against an empty database!